superset-frontend/src/components/Menu/MenuObject.tsx

```diff
@@ -36,16 +37,28 @@ export interface MenuObjectProps {
   url?: string;
   childs?: (MenuObjectChildProps | string)[];
   isHeader?: boolean;
+  frontEndRoutes?: {
+    [route: string]: boolean;
+  };
 }
```

Comment:
Bycatch: maybe let `MenuObject` extends from `MenuObjectChild`

````ts
export interface MenuObjectProps extends MenuObjectChildProps {
  isHeader?: boolean;
  childs?: (MenuObjectChildProps | string)[];
}
````

I'm also wondering whether it makes more sense to set `isFrontendRoute` to the menu object, instead of passing the full list of frontend routes around:

```ts
export interface MenuObjectChildProps {
  label: string;
  name?: string;
  icon: string;
  index: number;
  url?: string;
  isFrontendRoute?: boolean;
}
```

Reply:
I think one place or another we'll need to have a list of frontend routes, at least until all routes become assimilated.

Reply:
We could have a `routes.tsx` module where we keep the list of route & component pairs, along with other util functions like `isFrontendRoute`. That should help abstract away some of the stuff that's currently in the `src/views/App.tsx` module such as ErrorBoundary wrapping